### ISSUE TITLE
Missing source code (potential GPL violation)

### ISSUE THREAD

**Issue by andrigamerita** [44670/desmume-wasm#131]
Hello, sorry for the issue, but it seems to me that you probably forgot to update the source code on this repo over time!

The app on your linked website (ds.44670.org) says that it's been updated on the day of 2023-01-06, but your last commit here was on 2022-12-27. Actually, by looking through the commits I see that many parts have not been updated since even more time: 3 months for the actual DeSmuME source files, and 1 year for the wasm-port files.

This is a potential violation of the GPL-2 license, the license DeSmuME comes with, which requires that when you redistribute a modification or derived work (and your app certainly is) in binary form, you have to make the full source code available. A web application, if containing compiled parts like yours (nds.js and nds.wasm) and not being simply readable as source (think simple handwritten HTML and JS only), is considered a binary redistribution, since it runs on the client; GPL 2 not being Affero thus doesn't count in this case, and would only matter if your emulator was actually being streamed from a server.

If this was simply a mistake, you should remember to update your repo as soon as possible, and in general remember to make the source available as soon as you publish binaries.

**Comment by 44670** [comment]
Hi, the desmume CPP source code(nds.js nds.wasm files you mentioned) is not Ver. 2023-01-26, the Javascript source code is updated more frequently in these months, which the original source code is hosted on the server and you can simple download it directly.

For the updated javascript source code which is available on the server, it utilizes techniques like service worker and may have some issues if being published locally without https. That's why these code are not updated on the github, but you can always download them on the server, it is the the same version as the written version.

Thanks for pointing out, I will do a check on the affected CPP sources, I think only a few files s are waiting to be commited.

**Comment by 44670** [comment]
Hi, the updated CPP sources files are committed and uploaded. 

To avoid the misunderstanding that the latest HTML/JS files are not provided, the outdated JS copy in this repo has been removed and the user is instructed to go to the ds.44670.org server to download the latest HTML/JS source program (we deployed the original handwritten version directly on the server).

By the way, the original desmume contains a potentially copyrightable file that I had removed a long time ago. I took the opportunity to reset the entire repo so that it is not included in the commit log to avoid infringement.

**Comment by 44670** [comment]
Thanks for testing! Which lines in the main.cpp and app.js need to be fixed?

For now my build script may reproducible, and the original  work.

**Comment by andrigamerita** [comment]
I've not tested Dropbox, only that a ROM loads from storage, runs, and the savefile persists after the emulator tab is closed.
In `main.cpp`, line 134 (in `int main()` was giving a compilation error due to undeclared value, I simply prefixed "int" to it:
```c++
int eaVCode = rand() & 0x0FFFFFFF;
```
In app.js, 2 lines call internal functions that are missing from the newly compiled `nds.js` (but are present in that hosted on your website), at lines 330 and 442:
```js
// ...
/* 330 */ Module._setSampleRate(47856)
// ...
/* 442 */ Module._emuSetOpt(0, config.fwLang)
// ...
```
 I deleted them, and the game runs. Sound seems to work fine, so I don't think the first line is really needed, but changing the system language in the emulator settings (before starting a game) doesn't do anything, so I guess that the second line is needed.. only problem is the function doesn't exist!
